### DISCUSSION TITLE
ANSWERED - New person - looking for help with enabling CUDA.

### DISCUSSION THREAD

**Discussion by monoclyne** [ultralytics/ultralytics#4686] [Q&A]
I'm new and trying to train a YoloV8 Model on my own custom dataset. The specific computer that I have is an MSI Ge75 Raider, utilizing an RTX2060. I'm trying to use my RTX2060 by appending device=0 to the end of my bat command, but it just spits back that CUDA is not available. I already have the Nvidia CUDA Toolkit installed. I'm using the Jupyter plugin for Visual Studio Code, and I can't find anywhere to enable hardware acceleration.

I ran these commands:

```
!nvcc --version

Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Jun_13_19:42:34_Pacific_Daylight_Time_2023
Cuda compilation tools, release 12.2, V12.2.91
Build cuda_12.2.r12.2/compiler.32965470_0
Thu Aug 31 19:30:19 2023       


!nvidia-smi

+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 536.25                 Driver Version: 536.25       CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                     TCC/WDDM  | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 2060      WDDM  | 00000000:01:00.0  On |                  N/A |
| N/A   50C    P5              15W / 115W |   5373MiB /  6144MiB |      6%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      4932    C+G   [D:\Microsoft](file:///D:/Microsoft) VS Code\Code.exe             N/A      |
|    0   N/A  N/A     11216    C+G   ...CBS_cw5n1h2txyewy\TextInputHost.exe    N/A      |
...
|    0   N/A  N/A     27524    C+G   ...crosoft\Edge\Application\msedge.exe    N/A      |
+---------------------------------------------------------------------------------------+


!pip list | grep torch

torch                  2.0.1
torchvision            0.15.2


import torch as pt
pt.cuda.is_available()


False

```

Thank you!

EDIT:
    I figured it out. All I needed to do was install the version of PyTorch GPU version. I forgot to include the --upgrade modifier!
Original line from the PyTorch site: pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
Line I used to actually upgrade PyTorch: pip3 install --upgrade torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118

**Comment by glenn-jocher** [comment]
@Pagzma09 great to hear that you've resolved the issue! 👍 It's a common oversight to miss the `--upgrade` flag when installing packages. Ensuring that you have the correct version of PyTorch that matches your CUDA version is crucial for GPU acceleration. Now that you've got the right setup, you should be able to train your YOLOv8 model on your custom dataset using your RTX2060. If you encounter any further issues or have questions about using YOLOv8, feel free to reach out. Happy training! 😊
